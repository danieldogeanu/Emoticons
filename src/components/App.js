import React, { Component } from 'react';
import IconsSprite from './IconsSprite';
import Header from './Header';
import SearchBar from './SearchBar';
import EmoticonsTable from './EmoticonsTable';
import Footer from './Footer';
import './App.scss';

class App extends Component {
	render() {
		return (
			<div className="App">
				<IconsSprite />
				<Header />
				<SearchBar />
				<EmoticonsTable />
				<Footer />
			</div>
		);
	}
}

export default App;
