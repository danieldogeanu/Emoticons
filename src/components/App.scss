@import '../styles/includes';

.App {
	display: block;
	width: 100%; height: 100%;
	margin: rem(70) 0 0 0;
	@include breakpoint(min, rem(301)) { margin: rem(70) 0 0 0; }
	@include breakpoint(min, rem(481)) { margin: rem(90) 0 0 0; }
	@include breakpoint(min, rem(601)) { margin: rem(100) 0 0 0; }
}