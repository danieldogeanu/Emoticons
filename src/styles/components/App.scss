@import '../includes';

.App {
	display: block;
	width: 100%; height: 100%;
	padding: rem(70) 0 0 0;
	@include breakpoint(min, rem(301)) { padding: rem(70) 0 0 0; }
	@include breakpoint(min, rem(481)) { padding: rem(90) 0 0 0; }
	@include breakpoint(min, rem(601)) { padding: rem(100) 0 0 0; }
}