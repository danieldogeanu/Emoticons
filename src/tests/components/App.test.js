import React from 'react';
import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../../components/App';

jest.mock('../../emoji.json', () => [{
	"codes": "1F600",
	"char": "😀",
	"name": "grinning face",
	"category": "Smileys & Emotion (face-smiling)",
	"group": "Smileys & Emotion",
	"subgroup": "face-smiling"
}, {
	"codes": "1F603",
	"char": "😃",
	"name": "grinning face with big eyes",
	"category": "Smileys & Emotion (face-smiling)",
	"group": "Smileys & Emotion",
	"subgroup": "face-smiling"
}, {
	"codes": "1F604",
	"char": "😄",
	"name": "grinning face with smiling eyes",
	"category": "Smileys & Emotion (face-smiling)",
	"group": "Smileys & Emotion",
	"subgroup": "face-smiling"
}, {
	"codes": "1F601",
	"char": "😁",
	"name": "beaming face with smiling eyes",
	"category": "Smileys & Emotion (face-smiling)",
	"group": "Smileys & Emotion",
	"subgroup": "face-smiling"
}, {
	"codes": "1F606",
	"char": "😆",
	"name": "grinning squinting face",
	"category": "Smileys & Emotion (face-smiling)",
	"group": "Smileys & Emotion",
	"subgroup": "face-smiling"
}], {virtual: true});

const testEmoji = [
	{	"char": "😀", "name": "grinning face"},
	{	"char": "😃", "name": "grinning face with big eyes"},
	{	"char": "😄", "name": "grinning face with smiling eyes"},
	{	"char": "😁", "name": "beaming face with smiling eyes"},
	{	"char": "😆", "name": "grinning squinting face"}
];
const testFilterText = 'beaming';

const compNames = {
	app: 'App',
	sprite: 'IconsSprite',
	topBar: 'TopBar',
	search: {
		bar: 'SearchBar',
		input: 'SearchInput',
	},
	list: {
		container: 'ListContainer',
		item: {
			desktop: 'DesktopListItem',
			mobile: 'MobileListItem',
		},
	},
	scrollUp: 'ScrollUp',
	shortcuts: {
		button: 'ShortcutsButton',
		screen: 'ShortcutsScreen',
	},
	selected: 'SelectedScreen',
};

describe('App Component', () => {

	it('renders app properly', () => {
		const {getByTestId, getAllByTestId, container} = render(<App />);
		const rendered = {
			app: getByTestId(compNames.app),
			sprite: getByTestId(compNames.sprite),
			topBar: getByTestId(compNames.topBar),
			searchBar: getByTestId(compNames.search.bar),
			listContainer: getByTestId(compNames.list.container),
			listItems: getAllByTestId(compNames.list.item.desktop),
			scrollUp: container.querySelector(`.${compNames.scrollUp}`),
			shortcutsButton: container.querySelector(`.${compNames.shortcuts.button}`),
			shortcutsScreen: getByTestId(compNames.shortcuts.screen),
			selectedScreen: getByTestId(compNames.selected),
		};

		expect(rendered.app).toBeInTheDocument();
		expect(rendered.app).toHaveClass(compNames.app);
		expect(rendered.app).toContainElement(rendered.sprite);
		expect(rendered.app).toContainElement(rendered.topBar);
		expect(rendered.app).toContainElement(rendered.searchBar);
		expect(rendered.app).toContainElement(rendered.listContainer);
		expect(rendered.app).toContainElement(rendered.scrollUp);
		expect(rendered.app).toContainElement(rendered.shortcutsButton);
		expect(rendered.app).toContainElement(rendered.shortcutsScreen);
		expect(rendered.app).toContainElement(rendered.selectedScreen);
		expect(rendered.app).toMatchSnapshot();

		expect(rendered.listItems.length).toBe(5);
		rendered.listItems.forEach((item, i) => {
			expect(rendered.app).toContainElement(item);
			expect(item).toHaveTextContent(testEmoji[i].char);
			expect(item).toHaveTextContent(testEmoji[i].name);
		});
	});

	it('handles filter text change', async () => {
		const {getByTestId, getAllByTestId} = render(<App />);

		await userEvent.type(getByTestId(compNames.search.input), testFilterText);
		const rendered = {
			searchInput: getByTestId(compNames.search.input),
			listItems: getAllByTestId(compNames.list.item.desktop),
		};

		expect(rendered.searchInput.value).toBe(testFilterText);
		expect(rendered.listItems.length).toBe(1);
		expect(rendered.listItems[0]).toHaveTextContent(testFilterText);
	});

});
