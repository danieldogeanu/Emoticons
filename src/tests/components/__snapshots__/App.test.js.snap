// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders app properly 1`] = `
<div
  class="App"
  data-testid="App"
>
  <div
    class="IconsSprite"
    data-testid="IconsSprite"
    style="display: none;"
  >
    <svg>
      icons.svg
    </svg>
  </div>
  <div
    class="TopBar"
    data-testid="TopBar"
  >
    <div
      class="wrapper"
    >
      <form
        class="SearchBar"
        data-testid="SearchBar"
      >
        <input
          data-testid="SearchInput"
          placeholder="Search Emoticons..."
          type="text"
          value=""
        />
        <button
          class="ClearButton"
          title="Clear Input"
        >
          <span
            class="show-for-screen-reader"
            data-testid="ScreenReaderText"
          >
            Clear Input
          </span>
          <svg
            class="icon close"
          >
            <use
              xlink:href="#close"
            />
          </svg>
        </button>
      </form>
    </div>
  </div>
  <div
    class="ListContainer"
    data-testid="ListContainer"
  >
    <div
      class="List"
      data-simplebar="init"
      data-testid="List"
    >
      <div
        class="simplebar-wrapper"
      >
        <div
          class="simplebar-height-auto-observer-wrapper"
        >
          <div
            class="simplebar-height-auto-observer"
          />
        </div>
        <div
          class="simplebar-mask"
        >
          <div
            class="simplebar-offset"
            style="right: 0px; bottom: 0px;"
          >
            <div
              class="simplebar-content-wrapper"
              style="height: auto; overflow-x: hidden; overflow-y: hidden;"
            >
              <div
                class="simplebar-content"
              >
                <div
                  class="innerWrapper"
                >
                  <div
                    class="Labels"
                    data-testid="Labels"
                  >
                    <span
                      class="Label face"
                    >
                      Face
                    </span>
                    <span
                      class="Label name"
                    >
                      Emoticon Name
                    </span>
                    <span
                      class="Label copy"
                    >
                      Get
                    </span>
                  </div>
                  <ul
                    data-testid="listUl"
                    style="padding-top: 0px; height: 250px;"
                  >
                    <li
                      class="DesktopListItem"
                      data-testid="DesktopListItem"
                    >
                      <span
                        aria-label="grinning face"
                        class="Emoticon"
                        role="img"
                      >
                        üòÄ
                      </span>
                      <span
                        class="Name"
                      >
                        grinning face
                      </span>
                      <span
                        class="Copy"
                        title="Copy the emoticon and paste it where you need it."
                      >
                        <button
                          class="CopyButton undefined"
                          data-clipboard-text="üòÄ"
                          data-testid="CopyButton"
                          title=""
                          type="button"
                        >
                          Copy
                        </button>
                      </span>
                    </li>
                    <li
                      class="DesktopListItem"
                      data-testid="DesktopListItem"
                    >
                      <span
                        aria-label="grinning face with big eyes"
                        class="Emoticon"
                        role="img"
                      >
                        üòÉ
                      </span>
                      <span
                        class="Name"
                      >
                        grinning face with big eyes
                      </span>
                      <span
                        class="Copy"
                        title="Copy the emoticon and paste it where you need it."
                      >
                        <button
                          class="CopyButton undefined"
                          data-clipboard-text="üòÉ"
                          data-testid="CopyButton"
                          title=""
                          type="button"
                        >
                          Copy
                        </button>
                      </span>
                    </li>
                    <li
                      class="DesktopListItem"
                      data-testid="DesktopListItem"
                    >
                      <span
                        aria-label="grinning face with smiling eyes"
                        class="Emoticon"
                        role="img"
                      >
                        üòÑ
                      </span>
                      <span
                        class="Name"
                      >
                        grinning face with smiling eyes
                      </span>
                      <span
                        class="Copy"
                        title="Copy the emoticon and paste it where you need it."
                      >
                        <button
                          class="CopyButton undefined"
                          data-clipboard-text="üòÑ"
                          data-testid="CopyButton"
                          title=""
                          type="button"
                        >
                          Copy
                        </button>
                      </span>
                    </li>
                    <li
                      class="DesktopListItem"
                      data-testid="DesktopListItem"
                    >
                      <span
                        aria-label="beaming face with smiling eyes"
                        class="Emoticon"
                        role="img"
                      >
                        üòÅ
                      </span>
                      <span
                        class="Name"
                      >
                        beaming face with smiling eyes
                      </span>
                      <span
                        class="Copy"
                        title="Copy the emoticon and paste it where you need it."
                      >
                        <button
                          class="CopyButton undefined"
                          data-clipboard-text="üòÅ"
                          data-testid="CopyButton"
                          title=""
                          type="button"
                        >
                          Copy
                        </button>
                      </span>
                    </li>
                    <li
                      class="DesktopListItem"
                      data-testid="DesktopListItem"
                    >
                      <span
                        aria-label="grinning squinting face"
                        class="Emoticon"
                        role="img"
                      >
                        üòÜ
                      </span>
                      <span
                        class="Name"
                      >
                        grinning squinting face
                      </span>
                      <span
                        class="Copy"
                        title="Copy the emoticon and paste it where you need it."
                      >
                        <button
                          class="CopyButton undefined"
                          data-clipboard-text="üòÜ"
                          data-testid="CopyButton"
                          title=""
                          type="button"
                        >
                          Copy
                        </button>
                      </span>
                    </li>
                  </ul>
                  <footer
                    class="Footer"
                    data-testid="Footer"
                  >
                    <p
                      class="author"
                      data-testid="author"
                    >
                      By: 
                      <a
                        data-testid="Link"
                        href="https://danieldogeanu.com"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Daniel Dogeanu
                      </a>
                    </p>
                    <p
                      class="mynets"
                      data-testid="mynets"
                    >
                      <span
                        class="folw"
                      >
                        Follow Me: 
                      </span>
                      <span
                        class="wrapper"
                        data-testid="wrapper"
                      >
                        <a
                          data-testid="Link"
                          href="https://twitter.com/danieldogeanu"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <span
                            class="mynet"
                            data-testid="mynet"
                          >
                            <span
                              class="show-for-screen-reader"
                              data-testid="ScreenReaderText"
                            >
                              Twitter
                            </span>
                            <svg
                              class="icon twitter"
                            >
                              <use
                                xlink:href="#twitter"
                              />
                            </svg>
                          </span>
                        </a>
                        <a
                          data-testid="Link"
                          href="https://github.com/danieldogeanu"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <span
                            class="mynet"
                            data-testid="mynet"
                          >
                            <span
                              class="show-for-screen-reader"
                              data-testid="ScreenReaderText"
                            >
                              GitHub
                            </span>
                            <svg
                              class="icon github"
                            >
                              <use
                                xlink:href="#github"
                              />
                            </svg>
                          </span>
                        </a>
                        <a
                          data-testid="Link"
                          href="https://www.linkedin.com/in/danieldogeanu/"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <span
                            class="mynet"
                            data-testid="mynet"
                          >
                            <span
                              class="show-for-screen-reader"
                              data-testid="ScreenReaderText"
                            >
                              LinkedIn
                            </span>
                            <svg
                              class="icon linkedin"
                            >
                              <use
                                xlink:href="#linkedin"
                              />
                            </svg>
                          </span>
                        </a>
                        <a
                          data-testid="Link"
                          href="https://behance.net/danieldogeanu"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <span
                            class="mynet"
                            data-testid="mynet"
                          >
                            <span
                              class="show-for-screen-reader"
                              data-testid="ScreenReaderText"
                            >
                              Behance
                            </span>
                            <svg
                              class="icon behance"
                            >
                              <use
                                xlink:href="#behance"
                              />
                            </svg>
                          </span>
                        </a>
                      </span>
                    </p>
                  </footer>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="simplebar-placeholder"
          style="width: 0px; height: 0px;"
        />
      </div>
      <div
        class="simplebar-track simplebar-horizontal"
        style="visibility: hidden;"
      >
        <div
          class="simplebar-scrollbar"
          style="width: 0px; display: none;"
        />
      </div>
      <div
        class="simplebar-track simplebar-vertical"
        style="visibility: hidden;"
      >
        <div
          class="simplebar-scrollbar"
          style="height: 0px; display: none;"
        />
      </div>
    </div>
  </div>
  <button
    class="ScrollUp"
    title="Scroll Back Up"
  >
    <span
      class="show-for-screen-reader"
      data-testid="ScreenReaderText"
    >
      Scroll Back Up
    </span>
    <svg
      class="icon chevron-up"
    >
      <use
        xlink:href="#chevron-up"
      />
    </svg>
  </button>
  <button
    class="ShortcutsButton"
    title="Keyboard Shortcuts"
  >
    <span
      class="show-for-screen-reader"
      data-testid="ScreenReaderText"
    >
      Keyboard Shortcuts
    </span>
    <svg
      class="icon keyboard"
    >
      <use
        xlink:href="#keyboard"
      />
    </svg>
  </button>
  <div
    class="ShortcutsScreen"
    data-testid="ShortcutsScreen"
  >
    <div
      class="shortcutsContainer"
      data-testid="shortcutsContainer"
    >
      <span
        class="shortcutsTitle"
        data-testid="shortcutsTitle"
      >
        Keyboard Shortcuts
      </span>
      <ul
        class="shortcutsList"
        data-testid="shortcutsList"
      >
        <li
          data-testid="shortcutsKey"
        >
          <kbd>
            s
          </kbd>
          Selects Search Input
        </li>
        <li
          data-testid="shortcutsKey"
        >
          <kbd>
            esc
          </kbd>
          Clears Search Input / Exits Menu
        </li>
        <li
          data-testid="shortcutsKey"
        >
          <kbd>
            k
          </kbd>
          Opens Keyboard Shortcuts Screen
        </li>
        <li
          data-testid="shortcutsKey"
        >
          <kbd>
            t
          </kbd>
          Scrolls Back Up
        </li>
      </ul>
      <button
        class="ShortcutsClose"
        title="Close Shortcuts Screen"
      >
        <span
          class="show-for-screen-reader"
          data-testid="ScreenReaderText"
        >
          Close Shortcuts Screen
        </span>
        <svg
          class="icon close"
        >
          <use
            xlink:href="#close"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="SelectedScreen"
    data-testid="SelectedScreen"
  />
</div>
`;
